beaconChain/lib/stateTransition/block/proposerSignature.ts

```diff
@@ -0,0 +1,45 @@
+import assert from "assert";
+
+import {treeHash} from "@chainsafesystems/ssz";
+
+import {
+  BeaconBlock,
+  BeaconState,
+  bytes32,
+  ProposalSignedData,
+} from "../../../types";
+
+import {
+  BEACON_CHAIN_SHARD_NUMBER,
+  Domain,
+  EMPTY_SIGNATURE,
+} from "../../../constants";
+
+import {
+  getBeaconProposerIndex,
+  getCurrentEpoch,
+  getDomain,
+} from "../../../helpers/stateTransitionHelpers";
+
+import {blsVerify} from "../../stubs/bls";
+
+export default function processProposerSignature(state: BeaconState, block: BeaconBlock) {
+  block.signature = EMPTY_SIGNATURE;
+  const blockWithoutSignatureRoot: bytes32 = treeHash(block);
```

Comment:
Yeah, this part felt iffy. I think we need to keep the signature around, and not permanently modify the block.
something like:
```typescript
const signature = block.signature;
// temporarily set the signature as empty
block.signature = EMPTY_SIGNATURE;
const blockWithoutSignatureRoot: bytes32 = treeHash(block);
// reset the real signature
block.signature = signature;
```

Reply:
Yeah i think thats fine... it seems weird to remove the signature.
A deep clone instead of removing and re-adding might look cleaner, but then again might add a lot of time complexity...